specs/latest/2.0/index.html

```diff
@@ -1695,6 +1695,27 @@ <h4>Texture objects</h4>
       </dt>
       <dd>
         <p>Only differences from <a href="../1.0/index.html#TEXIMAGE2D_HTML">texImage2D</a> in WebGL 1.0 are described here. </p>
+        <p>Conversion to new formats introduced in WebGL 2.0 is performed according to the following table.</p>
+        <table class="foo">
+          <tr>
+            <th rowspan="2">Source DOM Image Format</th>
+            <th colspan="2">Target WebGL Format</th>
+          </tr>
+          <tr>
+            <th>RED</th>
+            <th>RG</th>
+          </tr>
+          <tr>
+            <td>Grayscale (1 channel)<br>Grayscale + Alpha (2 channels)</td>
+            <td>R = <em>sourceGray</em><br></td>
+            <td>R = <em>sourceGray</em><br>G = <em>sourceGray</em></td>
```

Comment:
G should be sourceAlpha. Don't be too literal about "RG", it's just a 2-channel format. :)

Reply:
If that's the intended behavior, then browsers do it wrong.
Looks like they expand grayscale source to color and then sample R and G from RGB image.

Reply:
I think we should do the thing I said, and I don't want to make the spec match behavior that we might not want.
Hopefully we'll decide on a direction on the conference call tomorrow.

Reply:
Just for reference, here's [Chromium DOM-to-RG8 conversion](https://github.com/chromium/chromium/blob/db29a96e938ab285182c7da5c8b4bd4a9b3c3934/third_party/blink/renderer/platform/graphics/gpu/webgl_image_conversion.cc#L1678). 
At that point, `source[]` has 4 components regardless of original image mode.

Firefox behaves the same way. A simple repro:
```html
<!DOCTYPE html>
<html>
<body onload="init()">
<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAADklEQVR4AQEDAPz/AP8AAgEBAIlSUwAAAAAASUVORK5CYII=" style="display: none", id='img'>
<canvas id="cnv" width="64" height="64" style="border: 1px solid black"></canvas>
<script>
    function init() {
      var cnv = document.getElementById('cnv');
      var img = document.getElementById('img');
      var gl = cnv.getContext('webgl2', {antialias: false});

      var tex = gl.createTexture();
      gl.bindTexture(gl.TEXTURE_2D, tex);
      gl.texImage2D(gl.TEXTURE_2D, 0, gl.RG8, 1, 1, 0, gl.RG, gl.UNSIGNED_BYTE, img);

      var fb = gl.createFramebuffer();
      gl.bindFramebuffer(gl.READ_FRAMEBUFFER, fb);
      gl.framebufferTexture2D(gl.READ_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, tex, 0);
      gl.readBuffer(gl.COLOR_ATTACHMENT0);

      gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, null);
      gl.drawBuffers([gl.BACK]);

      gl.blitFramebuffer(0, 0, 1, 1, 0, 0, 64, 64, gl.COLOR_BUFFER_BIT, gl.NEAREST);
    }
</script>
</body>
</html>
```
Supplied image contains just one pixel with Gray=255 and Alpha=0. Both Chrome and Firefox render it as yellow.

Reply:
@jdashg Has the WG been able to decide on a direction here?

Looking at this issue again, I don't think that treating Grayscale+Alpha images just as 2-channel format is consistent with RGB and RGBA mappings in WebGL 1.0 implementations.

Compare:
- to-RGB: R=G=B=_Gray_, _Alpha_ is lost
- to-RGBA: R=G=B=_Gray_, A=_Alpha_

Reply:
> Is there such a thing as e.g. a red-green PNG image?

No. PNG modes are RGB, RGBA, Grayscale, GrayscaleAlpha. JPEG modes are RGB and Grayscale.

> The rationale is that otherwise the two-channel image formats are fairly useless.

For red-green textures one could use RGB images with zeros in blue channel. They can be uploaded to gl.RG targets just fine.

> Do you have any feedback on this proposal?

I cannot think of a content pipeline that would put the second color channel into alpha. Also, it's not clear how premultiplication flag should affect such uploads.